### PR TITLE
Add tests to `UserList.cy.js`

### PR DESCRIPTION
**Pull request by astridx**
I have added a test to the file `UserList.cy.js` so that at the end the initial state is restored. Among other things, this has the advantage that you can repeat the test without reinstalling it and that there are no dependencies on other tests.

### CODE REVIEW

File: tests/cypress/integration/administrator/components/com_users/UserList.cy.js [hunk 1]
```diff
@@ -10,10 +10,9 @@ describe('Test com_users features', () => {
       "name": "Test Bot",
       "email": "Testbot@example.com"
     }
-
   })
 
-  it('creates a new user', function () {
+  it('disable sending mails', function () {
```
Comment (richard67) [inline tests/cypress/integration/administrator/components/com_users/UserList.cy.js]:
```suggestion
  it('disables sending mails', function () {
```

File: tests/cypress/integration/administrator/components/com_users/UserList.cy.js [hunk 2]
```diff
@@ -76,4 +77,43 @@ describe('Test com_users features', () => {
     cy.get('#system-message-container').contains('User saved.').should('exist')
     cy.checkForPhpNoticesOrWarnings()
   })
+
+  it('delets a user', function () {
```
Comment (richard67) [inline tests/cypress/integration/administrator/components/com_users/UserList.cy.js]:
```suggestion
  it('deletes a user', function () {
```

File: tests/cypress/integration/administrator/components/com_users/UserList.cy.js [hunk 3]
```diff
@@ -76,4 +77,43 @@ describe('Test com_users features', () => {
...
+
+    cy.on("window:confirm", (s) => {
+      return true;
+    });
+      
+    cy.get('#system-message-container').contains('User deleted.').should('exist')
+    cy.checkForPhpNoticesOrWarnings()
+  })
+
+  it('enable sending mails', function () {
```
Comment (richard67) [inline tests/cypress/integration/administrator/components/com_users/UserList.cy.js]:
```suggestion
  it('enables sending mails', function () {
```

**Comment by laoneo** [comment]
@astridx are you going to add more tests? Otherwise I would merge it.

**Comment by astridx** [comment]
> @astridx are you going to add more tests? Otherwise I would merge it.

You ask this because I named the commit `wip`? I am just taking my first steps with Cypress. I don't yet have an overview of how to integrate it usefully into Joomla. Are there other sources besides 

- https://magazine.joomla.org/all-issues/october-2022/off-to-cyprus-ehm-cypress-how-joomla-does-its-end-to-end-testing
- https://github.com/joomla/joomla-cms/pull/38422
- https://www.youtube.com/watch?v=26jL9EVI-98

  
If my PR makes sense for Joomla, I'm happy to merge it.

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**